### ISSUE TITLE
No balanced split 

### ISSUE THREAD

**Issue by jaimerodric** [ma7555/kerasgen#4]
Hi! After training with _balanced_image_dataset_from_directory_ , I have tried to test the validation dataset that your app made.
What I get from it is the following:

> Found 161866 files belonging to 9198 classes.
> Using 129493 files for training.
> Found 161866 files belonging to 9198 classes.
> Using 32373 files for validation.

I thought that every class would be used in validation database, but what I get using:
` x = np.concatenate([x for x, y in val_ds], axis=0)`
`y = np.concatenate([y for x, y in val_ds], axis=0)`

The variable y has just 4500 unique classes. Where are the ones missing of those 9198 classes? It's not about not having enough photos because the class 3 (for example, one missing in validation dataset), it has more than 10 photos. The parameters of my balanced_image_dataset are these ones:

>     val_ds = balanced_image_dataset_from_directory(
>         directory, num_classes_per_batch=64,
>         num_images_per_class=4, image_size=(160, 160),
>         seed=6, validation_split=0.2, subset='validation',
>         safe_triplet=True)

**Comment by ma7555** [comment]
I apologize for the "very" late reply. I would need more details on your directory structure in order to investigate. There are better alternatives than kerasgen now using [TFDataSampler](https://github.com/tensorflow/similarity/blob/dc506d63d05d012d3496041c25c1c54981d39565/tensorflow_similarity/samplers/tfdata_sampler.py#L135) from TF-Similarity which can be wrapped over the normal image_dataset_from_directory.
